app/components/Swapbox.tsx

```diff
@@ -362,31 +370,20 @@ export const Swapbox = ({ fixedProductMarketMaker }: SwapboxProps) => {
               </div>
             )}
           </div>
-          {isDisconnected ? (
-            <ConnectButton width="full" size="lg">
-              Connect
-            </ConnectButton>
-          ) : unsupportedNetwork ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Unsupported network
-            </Button>
-          ) : !tokenAmountIn || +tokenAmountIn === 0 ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Enter amount
-            </Button>
-          ) : currentState.isLoading ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Fetching price
-            </Button>
-          ) : +tokenAmountIn > +formatEther(currentState.balance) ? (
-            <Button width="full" variant="pastel" size="lg" disabled>
-              Insufficient {currentState.inToken.symbol} balance
-            </Button>
-          ) : (
-            <Button width="full" variant="pastel" size="lg" onClick={openBetModal}>
-              {currentState.buttonText}
-            </Button>
-          )}
+          <AccountStateButton
+            disabled={isButtonDisabled}
+            onClick={openBetModal}
+            size="lg"
+            variant="pastel"
+            width="full"
+          >
+            {getButtonLabel(
+              +tokenAmountIn,
+              defaultButtonLabel,
+              isEnoughBalance,
+              tokenSymbolText
+            )}
```

Comment:
I think it's better to have the condition logic here in the component.
It can be inside a function, but should be inside the component.

Let's not abstract this logic for now.